modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js

```diff
@@ -34,73 +34,71 @@ const Overview = ({project, subscriptionGroups}) => {
 		SUBSCRIPTIONS_STATUS.future,
 	]);
 
+	const [allSubscriptions, setAllSubscriptions] = useState([]);
+
+	const [
+		subscriptionGroupsWithSubscriptions,
+		setSubscriptionGroupsWithSubscriptions,
+	] = useState([]);
+
 	const parseAccountSubscriptionGroupERC = (subscriptionName) => {
 		return subscriptionName.toLowerCase().replace(' ', '-');
 	};
 
-	const getAccountSubscriptionFilterQueryString = (
-		previousValue,
-		currentValue,
-		currentIndex,
-		array
-	) => {
-		if (currentIndex === array.length - 1) {
-			return previousValue + ` subscriptionStatus eq '${currentValue}'`;
-		}
-
-		return (
-			previousValue +
-			` subscriptionStatus eq '${currentValue}' or accountSubscriptionGroupERC eq '${
-				project.accountKey
-			}_${parseAccountSubscriptionGroupERC(
-				selectedSubscriptionGroup
-			)}' and`
-		);
-	};
+	useEffect(() => {
+		const subscriptionsToRender = allSubscriptions
+			.filter(
+				(subscription) =>
+					subscription.accountSubscriptionGroupERC.replace(
+						`${project.accountKey}_`,
+						''
+					) ===
+					parseAccountSubscriptionGroupERC(selectedSubscriptionGroup)
+			)
+			.filter((subscription) =>
+				selectedStatus.includes(subscription.subscriptionStatus)
+			);
+
+		setAccountSubscriptions(subscriptionsToRender);
+	}, [allSubscriptions, project, selectedStatus, selectedSubscriptionGroup]);
 
 	useEffect(() => {
-		const getSubscriptions = async (
-			accountKey,
-			subscriptionGroup,
-			status
-		) => {
+		const getAllSubscriptions = async (accountKey) => {
 			const {data: dataAccountSubscriptions} = await client.query({
 				query: getAccountSubscriptions,
 				variables: {
-					filter: `accountSubscriptionGroupERC eq '${accountKey}_${parseAccountSubscriptionGroupERC(
-						subscriptionGroup
-					)}'${
-						status.length ===
-						Object.keys(SUBSCRIPTIONS_STATUS).length
-							? ''
-							: `${status.reduce(
-									getAccountSubscriptionFilterQueryString,
-									' and'
-							  )}`
-					}`,
+					filter: `accountKey eq '${accountKey}'`,
 				},
 			});
 
 			if (dataAccountSubscriptions) {
-				setAccountSubscriptions(
-					dataAccountSubscriptions?.c?.accountSubscriptions?.items
+				const dataAllSubscriptions =
+					dataAccountSubscriptions?.c?.accountSubscriptions?.items;
+
+				const accountSubscriptionGroups = subscriptionGroups.filter(
+					(subscriptionGroup) =>
+						dataAllSubscriptions.some(
+							(subscription) =>
+								subscription.accountSubscriptionGroupERC.replace(
+									`${accountKey}_`,
+									''
+								) ===
+								parseAccountSubscriptionGroupERC(
+									subscriptionGroup.name
+								)
+						)
+				);
+
+				setAllSubscriptions(dataAllSubscriptions);
+
+				setSubscriptionGroupsWithSubscriptions(
+					accountSubscriptionGroups
```

Comment:
Convert this useEffect to an useMemo. return an object with the accountSubscriptionGroups and the dataAllsubscriptions
```suggestion
			return {
			accountSubscriptionGroups,
			allSubscriptions: dataAllSubscriptions
			}
```